File: wallet.js
const prompt = require("prompt-sync")()

let defaultUsername = "Dinasoa"
let defaultPassword = "averystrongpassword" 
let solde = 0;
let transactions = []
let monthlyBudget = 200000

const showBalance = () => {
    console.log(`Votre solde actuel: ${solde} Ar`)
}

const deposit = () => {
    const amount = parseFloat(prompt("Veuillez saisir le montant à déposer: Ar"));
    if(!isNaN(amount) && amount > 0){
        solde += amount;
        transactions.push(`Dépot de ${amount} Ar`)
        console.log(`Dépot effectué! `)
    } else {
        console.log("Le montant que vous avez saisi est invalide, veuillez verifier! ")
    }
}

const withdraw = () => {
    const amount = parseFloat(prompt("Veuillez saisir le montant à retirer: Ar"));
    if(!isNaN(amount) && amount > 0){
        if(amount <= solde){
            solde -= amount
            transactions.push(`Retrait de ${amount} Ar. `)
            console.log("Retrait effectué. ")
        } else {
            console.log("Votre solde est insuffisant. ")
        }
    } else {
        console.log("Montant invalide. ");
    }
}

const showTransactions = () => {
    console.log("Historique de transactions: ")
    for (const transaction of transactions) {
        console.log(transaction);
    }
}

const setMonthlyBudget = () => {
    const budget = parseFloat(prompt("Veuillez saisir votre nouveau budget mensuel: Ar"))
    if(!isNaN(budget) && budget > 0){
        monthlyBudget = budget;
        console.log(`Votre budget mensuel a été mis à jour: ${monthlyBudget} Ar`);
    }else{
        console.log("Montant invalide. ")
    }
}

const main = () => {
    let username = prompt("Veuillez entrer votre nom d'utilisateur: ");
    let userPassword = prompt("Veuillez confirmer votre identité en tapant le mot de passe: ");

    if(username != defaultUsername || userPassword != defaultPassword){
        console.log("Oups, vous n'avez pas acces à ce wallet, sorry. ")
    } else {
        let choice;

        do {
                console.log(`Bienvenue dans "Wallet" ${defaultUsername}, veuillez choisir une action à effectuer:
                1- Voir le solde dans mon compte
                2- Faire un dépôt
                3- Faire un retrait
                4- Afficher l'historique des transactions
                5- Mettre à jour le budget mensuel
                6- Quitter`);

                choice = +prompt("Votre choix: ");

                switch (choice) {
                case 1:
                    showBalance();
                    break;
                case 2:
                    deposit();
                    break;
                case 3:
                    withdraw();
                    break;
                case 4:
                    showTransactions();
                    break;
                case 5:
                    setMonthlyBudget();
                    break;
                case 6:
                    console.log("Merci d'utiliser le service Wallet. Au revoir!");
                    break;
                default:
                    console.log("Choix invalide. Veuillez choisir une option valide.");

                }
        } while (choice != 6)
    }

}

main()

File: wallet.py
default_username = "Dinasoa"
default_password = "averystrongpassword"
solde = 0
transactions = []
monthly_budget = 200000


def show_balance():
    print(f"Votre solde actuel: {solde} Ar. ")


def deposit():
    global solde
    amount = float(input("Veuillez saisir le montant à déposer: Ar "))
    try:
        if amount > 0:
            solde += amount
            transactions.append(f"Depot de {amount} Ar. ")
    except ValueError:
        print(f"Montan invalide, veuillez verifier! ")


def withdraw():
    global solde
    amount = float(input("Veuillez saisir le montant à retirer: Ar "))
    try:
        if amount <= solde:
            solde -= amount
            transactions.append(f"Retrait de {amount} Ar. ")
        else:
            print("Votre solde est insuffisant. ")
    except ValueError:
        print(f"Montan invalide, veuillez verifier! ")


def show_transactions():
    print(f"Historique des transactions: ")
    for transaction in transactions:
        print(transaction)


def set_monthly_budget():
    global monthly_budget
    budget = float(input("Veuillez saisir votre nouveau budget mensuel: Ar "))

    if budget > 0:
        monthly_budget = budget
        print(f"Votre budget mensuel a été mis à jour: {monthly_budget} Ar")
    else:
        print("Montant invalide. ")

if __name__ == '__main__':
    username = input(f"Veuillez entrer votre nom: ")
    user_password = input(f"Veuillez confirmer votre identité en tapant le mot de passe: ")

    if username != default_username or user_password != default_password:
        print(f"Oups, vous n'avez pas acces à ce wallet, sorry. ")
    else:
        choice = 0

        while choice != 6:
            print(f"Bienvenue dans 'Wallet' {default_username}, veuillez choisir une action à effectuer:"
                  "\n1- Voir le solde dans mon compte"
                  "\n2- Faire un dépôt"
                  "\n3- Faire un retrait"
                  "\n4- Afficher l'historique des transactions"
                  "\n5- Mettre à jour le budget mensuel"
                  "\n6- Quitter")

            choice = int(input("Votre choix: "))

            if choice == 1:
                show_balance()
            elif choice == 2:
                deposit()
            elif choice == 3:
                withdraw()
            elif choice == 4:
                show_transactions()
            elif choice == 5:
                set_monthly_budget()
            elif choice == 6:
                print(f"Merci d'utiliser le service Wallet. Au revoir! ")
            else:
                print("Choix invalide. Veuillez choisir une option valide. ")